### DISCUSSION TITLE
Programmatic way to launch Storybook

### DISCUSSION THREAD

**Discussion by azarenkovgd** [storybookjs/storybook#23128] [Help] [answered]
Is there any way to run storybook programmatically? 
This would be especially useful for mono repositories, where you want to install and configure storybook once, and then use it many times.

For example: 
```ts
import storybook from "@storybook/api"

storybook({
...some launch options that are usually passed to CLI
})
```
Many other js projects provide this capability, including webpack, vite and esbuild. 

I have studied issue https://github.com/storybookjs/storybook/issues/146, but there is no adequate answer to this question. I understand that you can run the storybook cli in nodejs manually in the shell, but it's not very convenient.

**Comment by shilman** [comment, marked as answer]
Not really a public API, but available from `@storybook/core-server`:

https://github.com/storybookjs/storybook/blob/054630104fc8198977754f640f20091c1dfc6f62/code/lib/core-server/src/build-dev.ts#L31-L33

https://github.com/storybookjs/storybook/blob/054630104fc8198977754f640f20091c1dfc6f62/code/lib/core-server/src/build-static.ts#L44

**Reply by azarenkovgd** [reply]
Thank you!
